be2-scala/docs/README.md

```diff
@@ -224,22 +224,117 @@ val f: Future[GraphMessage] = (dbActor ? DbActor.Write(channel, message)).map {
 	)
 }
 
-// Await.result waits for <duration> for the future <f> to complete. It returns the value contained by the future (here `GraphMessage`) if the latter is succesful, or throws if the Future termintes withtout being successful (i.e. either Failure or Timeout) 
+// Await.result waits for <duration> for the future <f> to complete. It returns the value contained by the future (here `GraphMessage`) if the latter is successful, or throws if the Future terminates without being successful (i.e. either Failure or Timeout) 
 Await.result(f, duration)
 ```
 
 
 
 :information_source: the database may easily be reset/purged by either deleting a folder corresponding to a channel or by deleting the `database` folder entirely
 
-## Debugging Tips
+## 4.	Git(Hub)
+
+### Import the project
+
+`cd` where you want to have your project located and then clone the project using the following commands
+
+```bash
+git clone https://github.com/dedis/student_21_pop.git [folderProjectName]
+```
+
+We then want to navigate to the `be2-scala` subfolder
+
+```bash
+cd <folderProjectName>/be2-scala
+```
+
+You can then open the project using your favorite IDE (e.g. `idea .` for IntelliJ and `code .` for VSCode)
+
+---
+
+### Create a new branch
+
+By default, the new branch you create will be a copy of `master`
+
+```bash
+git checkout -b <branchName>
+```
+
+If you rather want to make a copy of branch `x` (e.g. a branch that is currently being reviewed, i.e. not merged), use this instead
+
+```bash
+git checkout <x>
+git chekcout -b <branchName>
+```
+
+---
+
+### Push changes on the remote server
+
+When pushing changes, you first want to **commit** the changes you've applied using
+
+```bash
+git add -A									# add all modified and new files to the commit
+git commit -m"<message>"    # add a title to the commit (switch <message> for the actual message)
+
+## Note: You may also add files separately using
+git add <fileName>
+
+## Note: You may also add all modified (!= new) files to the commit and set a title using
+git commit -am"<message>"
+
+## Note: You may also set a title and description for a specific commit using your favorite IDE (obviously doom emacs ^^). Then save and quit
+git add -A
+git commit
+```
```

Comment:
I would avoid doing this as it's very easy to commit the wrong things. 
I would suggest looking at `git add -p` , `git add -u`, `git diff --cached` and `git status` to make sure that you know exactly what is being committed.